commons/types/account.go

```diff
@@ -276,6 +276,45 @@ func ToAccountsByName(name string, txn *badger.Txn) ([]*Account, error) {
 	return Accounts, nil
 }
 
+func AccountPaging(page int,txn *badger.Txn) ([]*Account, error){
+	var iteratorCount = 0
+	var firstItem int
+	if page == 0 {
+		return nil, ErrInvalidRequest
+	}else if page == 1{
+		firstItem = 1
+	} else{
+		firstItem = (page * 10) - 9
+	}
+
+	defer txn.Discard()
+	iterator := txn.NewIterator(badger.DefaultIteratorOptions)
+	defer iterator.Close()
+	prefix := []byte(fmt.Sprintf("table-account-"))
+	var Accounts = make([]*Account, 0)
+	for iterator.Seek(prefix); iterator.ValidForPrefix(prefix); iterator.Next() {
+		iteratorCount++
+		if iteratorCount >= firstItem && iteratorCount <= (firstItem+9) {
```

Comment:
Running through the Iterator in this way will get progressively slower as the page gets deeper. We should look into the Pager interface: https://godoc.org/google.golang.org/api/iterator#Pager

Reply:
Looking into this more, I don't think we'll be able to use Pager - the need to filter the prefix negates that ability. :-(